### PR TITLE
infosync, types: Truncate float for TiFlash replica progress

### LINKED ISSUE #24298: TiFlash replica progress rounded to 1 if the actual progress is no less than 0.995
When the TiFlash replica progress is no less than 0.995, e.g., in cases where there are many regions to be synchronized, if we use `select * from INFORMATION_SCHEMA.TIFLASH_REPLICA`, the result will be rounded to 1. 

It causes confusion.

### PR DESCRIPTION
**Pull request by birdstorm**
### What problem does this PR solve?

Issue Number: close #24298 

Problem Summary:

### What is changed and how it works?

What's Changed:

Add a truncate float API so that the replica progress will not be rounded to 1 when it actually isn't.

How it Works:

Truncate the progress result to 2 digits after the decimal point. 

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

**Comment by solotzg** [comment]
@crazycs520 PTAL

**Comment by crazycs520** [comment]
@AilinKid PTAL

File: types/helper_test.go [hunk 1]
```diff
@@ -63,3 +63,25 @@ func (s *testTypeHelperSuite) TestTruncate(c *C) {
...
+		f		 float64
+		dec		 int
+		expected string
+	}{
+		{12.13, -1, "10"},
+		{13.15, 0, "13"},
+		{0, 2, "0"},
+		{0.001, 2, "0"},
+		{0.539, 2, "0.53"},
+		{0.995, 2, "0.99"},
```
Comment (solotzg) [inline types/helper_test.go:79]: 0.995 will produce 0.99 in original logic. It's better to chose >0.995

**Review by AilinKid — COMMENTED** [review]
/lgtm

**Pull request merged**